ArduPlane/quadplane.cpp

```diff
@@ -3219,51 +3221,28 @@ float QuadPlane::get_weathervane_yaw_rate_cds(void)
     /*
       we only do weathervaning in modes where we are doing VTOL
       position control. We also don't do it if the pilot has given any
-      yaw input in the last 3 seconds.
+      yaw input in the last 2 seconds.
     */
     if (!in_vtol_mode() ||
+        tailsitter.in_vtol_transition() ||
         !motors->armed() ||
-        weathervane.gain <= 0 ||
         plane.control_mode == &plane.mode_qstabilize ||
 #if QAUTOTUNE_ENABLED
         plane.control_mode == &plane.mode_qautotune ||
 #endif
-        plane.control_mode == &plane.mode_qhover
+        plane.control_mode == &plane.mode_qhover ||
+        should_relax()
         ) {
-        weathervane.last_output = 0;
-        return 0;
-    }
-    const uint32_t tnow = millis();
-    if (plane.channel_rudder->get_control_in() != 0) {
-        weathervane.last_pilot_input_ms = tnow;
-        weathervane.last_output = 0;
-        return 0;
-    }
-    if (tnow - weathervane.last_pilot_input_ms < 3000) {
-        weathervane.last_output = 0;
-        return 0;
+        // Ensure the weathervane controller is reset to prevent weathervaning from happening outside of the timer
+        weathervane->reset();
+        return 0.0;
     }
 
-    float roll = wp_nav->get_roll() / 100.0f;
```

Comment:
We have lost this divide by 100, hence the gain is much higher.

Reply:
This isn't the issue with the gain.  The factor of 100 gets re-applied in the return of the function
https://github.com/ArduPilot/ardupilot/blob/f50f8c7a2a46a772312295e2ae03c0e78bd71418/ArduPlane/quadplane.cpp#L3279